### ISSUE TITLE
German lemmatization

### ISSUE THREAD

**Issue by uniquer4ven** [explosion/spaCy#1326]
Hi all,
I just tried spaCy with R (R, AnacondaPython, Win10 all have the version of last week), applying it to a German text. Everything went fine, including the POS tagging, which would not work, I guess, if the initialization failed, BUT: all the lemmas are as the original words, declension is always as in the original text. Do you have any solutions?
I checked with English, that worked fine, also a mistake in the R interface could be eliminated, which I am sure can be confirmed, when trying the same procedure in Python directly.

```
katzeHut<-c("Die Sonne schien nicht.", "Es war zu nass um zu spielen.", "Also saßen wir im Hause den ganzen bitterkalten, nassen Tag lang.")
obj <- run_annotators(katzeHut, as_strings = TRUE)
get_token(obj)
# A tibble: 26 × 8
      id   sid   tid   word  lemma  upos    pos   cid
   <int> <int> <int>  <chr>  <chr> <chr>  <chr> <int>
1      1     1     1    Die    die   DET    ART     0
2      1     1     2  Sonne  sonne  NOUN     NN     4
3      1     1     3 schien schien  VERB  VVFIN    10
4      1     1     4  nicht  nicht  PART PTKNEG    17
5      1     1     5      .      . PUNCT     $.    22
6      2     1     1     Es     es  PRON   PPER     0
7      2     1     2    war    war   AUX  VAFIN     3
8      2     1     3     zu     zu  PART   PTKA     7
9      2     1     4   nass   nass   ADJ   ADJD    10
10     2     1     5     um     um   ADP   APPR    15
# ... with 16 more rows
```

**Comment by honnibal** [comment]
Lemmatization isn't currently supported for German: https://spacy.io/docs/api/language-models

English lemmatization is very easy, because it's so morphologically simple. Other languages pose different problems. We're working on this for spaCy 2.

**Comment by uniquer4ven** [comment]
Thanks for the feedback, so what is the guesstimate for spaCy 2 then, roughly?

**Comment by ines** [comment]
@uniquer4ven We don't want to commit to an exact deadline because it depends on too many factors – but we're getting pretty close to a stable version, and the feedback on the alpha has been pretty good so far! To see what's still missing, you can check out the [v2.0.0 stable project board](https://github.com/explosion/spaCy/projects/4), organised by classes. The German model is currently training and we're hoping to have an alpha version up soon for people to test.

In the meantime, the current [v2.0.0 alpha release](https://github.com/explosion/spaCy/releases/tag/v2.0.0-alpha) already comes with a new [alternative lookup-based lemmatizer](https://github.com/explosion/spaCy/blob/develop/spacy/lemmatizerlookup.py) – it's not wired up yet as the default lemmatization solution for languages with no model available, but you can already play with it and use the German lookup table in [`lang/de/lemmatizer.py`](https://github.com/explosion/spaCy/blob/develop/spacy/lang/de/lemmatizer.py). (If you do, we're very curious to hear your feedback!)

**Comment by Liebeck** [comment]
@uniquer4ven I'm currently using https://github.com/Liebeck/IWNLP-py for German lemmatization. After spaCy has processed, I iterate over every token and retrieve possible lemmas. A couple of my colleagues and students are using it in combination with spaCy as well.

I'll focus on a more direct spaCy integration with the new pipeline model in 2.0 (see #1085). I'm currently stuck / waiting for German models for POS-tagging and parsing to be added to v2 so that it actually makes sense to test the direct integration ;) (The English model + German lemmatization doesn't make sense ^^)

**Comment by ines** [comment]
@uniquer4ven @Liebeck The basic lookup lemmatization is now implemented for all languages with a lookup table available, including German. Should be merged onto `develop` soon and will be included in the next alpha release – see PR #1413.
